core/iwasm/fast-jit/fe/jit_emit_variable.c

```diff
@@ -161,15 +161,152 @@ jit_compile_op_tee_local(JitCompContext *cc, uint32 local_idx)
     return false;
 }
 
+static uint8
+get_global_type(const WASMModule *module, uint32 global_idx)
+{
+    return module->globals[global_idx].type;
+}
+
+static JitReg
+get_data_offset_reg(JitCompContext *cc, uint32 global_idx)
+{
+    JitReg globals, global_inst, data_offset;
+
+    globals = get_globals_reg(cc->jit_frame);
+
+    /* global_inst = globals + global_idx */
+    global_inst = jit_cc_new_reg_ptr(cc);
+    GEN_INSN(ADD, global_inst, globals,
+             NEW_CONST(I64, global_idx * sizeof(WASMGlobalInstance)));
+
+    /* data_offset = global_inst->data_offset */
+    data_offset = jit_cc_new_reg_I64(cc);
+    GEN_INSN(LDI32, data_offset, global_inst,
+             NEW_CONST(I32, offsetof(WASMGlobalInstance, data_offset)));
+
+    return data_offset;
+}
+
 bool
 jit_compile_op_get_global(JitCompContext *cc, uint32 global_idx)
 {
+    uint8 global_type = 0;
+    JitReg value = 0;
+
+    if (global_idx >= cc->cur_wasm_module->import_global_count
+                          + cc->cur_wasm_module->global_count) {
```

Comment:
The global_idx has been checked in loader stage, how about using bh_assert instead?